pkg/deploy/cmd/delete.go

```diff
@@ -31,34 +33,36 @@ type DeploymentConfigReaper struct {
 	pollInterval, timeout time.Duration
 }
 
-type updateConfigFunc func(d *deployapi.DeploymentConfig)
-
-// updateConfigWithRetries will try to update a deployment config and ignore any update conflicts.
-func updateConfigWithRetries(dn appsinternal.DeploymentConfigsGetter, namespace, name string, applyUpdate updateConfigFunc) (*deployapi.DeploymentConfig, error) {
-	var config *deployapi.DeploymentConfig
-
-	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		var err error
-		config, err = dn.DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		// Apply the update, then attempt to push it to the apiserver.
-		applyUpdate(config)
-		config, err = dn.DeploymentConfigs(namespace).Update(config)
-		return err
-	})
-	return config, resultErr
-}
-
 // pause marks the deployment configuration as paused to avoid triggering new
 // deployments.
 func (reaper *DeploymentConfigReaper) pause(namespace, name string) (*deployapi.DeploymentConfig, error) {
-	return updateConfigWithRetries(reaper.appsClient.Apps(), namespace, name, func(d *deployapi.DeploymentConfig) {
-		d.Spec.RevisionHistoryLimit = kutil.Int32Ptr(0)
-		d.Spec.Replicas = 0
-		d.Spec.Paused = true
-	})
+	config, err := reaper.appsClient.Apps().DeploymentConfigs(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	encoder := kapi.Codecs.LegacyCodec(kapi.Registry.EnabledVersions()...)
+
+	oldConfigData, err := runtime.Encode(encoder, config)
+	if err != nil {
+		return nil, err
+	}
+
+	newConfig := *config
+	newConfig.Spec.RevisionHistoryLimit = kutil.Int32Ptr(0)
+	newConfig.Spec.Replicas = 0
+	newConfig.Spec.Paused = true
+
+	newConfigData, err := runtime.Encode(encoder, &newConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldConfigData, newConfigData, &deployapi.DeploymentConfig{})
+	if err != nil {
+		return nil, err
+	}
+	return reaper.appsClient.Apps().DeploymentConfigs(namespace).Patch(name, types.StrategicMergePatchType, patchBytes)
```

Comment:
shouldn't this use the external clientset and AppsV1() ?

Reply:
no, we are switching to internal clientset for now.

Reply:
Why? I though external clients should use external clientset...

Reply:
not yet :-) it will mean we have to refactor all helpers inside to switch to external version, internal version is an immediate step to external